internal/server/data/migrations.go

```diff
@@ -714,3 +715,30 @@ func addProviderUserSCIMFields() *migrator.Migration {
 		},
 	}
 }
+
+func addDeviceFlowAuthRequestTable() *migrator.Migration {
+	return &migrator.Migration{
+		ID: "2022-10-06T14:58",
+		Migrate: func(tx migrator.DB) error {
+			_, err := tx.Exec(`
+				CREATE TABLE IF NOT EXISTS device_flow_auth_requests (
+					id 							 bigint NOT NULL,
+					client_id 			 text NOT NULL,
+					user_code 			 text NOT NULL,
+					device_code 		 text NOT NULL,
+					approved 				 bool DEFAULT NULL,
+					access_key_id 	 bigint,
+					access_key_token text,
+					expires_at 			 timestamp with time zone,
+					created_at 			 timestamp with time zone,
+					updated_at 			 timestamp with time zone,
+					deleted_at 			 timestamp with time zone
+				);
+
+				CREATE UNIQUE INDEX IF NOT EXISTS idx_dfar_user_code on device_flow_auth_requests (user_code) WHERE (deleted_at IS NULL);
```

Comment:
This index should probably also include an org ID or device code, clashes could be frequent if we dont delete the device code after it expires.

Reply:
chances of collision are n in 20^8, where n is the number of existing records.
I think instead we'll just delete them frequently. working on adding something for that.